### ISSUE TITLE
Yet more `'$_AwaitThenValue1' is not defined`

### ISSUE THREAD

**Issue by marekdedic** [ota-meshi/eslint-plugin-svelte#338]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.23.0

### What version of `eslint-plugin-svelte` are you using?

2.14.1

### What did you do?

<details>
<summary>Configuration</summary>

```
{
  "parser": "svelte-eslint-parser",
  "parserOptions": {
    "parser": "@typescript-eslint/parser",
    "extraFileExtensions": [".svelte"]
  },
  "rules": {
    "no-undef": "error"
  }
}
```
</details>

```svelte
<script lang="ts">
  let lockedError: string | null = null;
  const mutexPromise = new Promise<void>();
</script>

{#await mutexPromise}
{:then _}
  {#if lockedError !== null}
  {:else}
  {/if}
{/await}
```

### What did you expect to happen?

No error

### What actually happened?

Yet more `'$_AwaitThenValue1' is not defined` - same as in #325 

### Link to **GitHub Repo** with Minimal Reproducible Example

Will provide if necessary

### Additional comments

I'm on `svelte-eslint-parser` 0.22.3

**Comment by ota-meshi** [comment]
> Will provide if necessary

A minimal reproduction is always required

**Comment by marekdedic** [comment]
Thanks for the quick fix!
